### DISCUSSION TITLE
Xonsh unexpectedly sourcing foreign shell run control files — Why?

### DISCUSSION THREAD

**Discussion by JeffMelton** [xonsh/xonsh#5764] [Q&A] [answered]
I stumbled on this when I added carapace to my Xonsh run control, and I opened an issue over there. Because there's still unexpected behavior from Xonsh without carapace in the mix, I'm also dropping a question here. [Caveat: I'm just getting started with Xonsh, slowly rewriting stuff from my Elvish daily driver, so I'm starting with the assumption that PEBKAC. Sprinkle a grain of salt on everything here because it's possible I've unwittingly footgunned myself.]

My `~/.config/xonsh` directory looks like this:
```
❯ tree ~/.config/xonsh
├── rc.d
│   ├── 00-path.xsh
│   ├── 01-xontribs.xsh
│   ├── <redacted>
│   ├── 06-env.xsh
│   ├── 10-k8s.xsh
│   ├── 15-util.xsh
│   ├── 20-ops.xsh
│   ├── 98-aliases.xsh
│   ├── 99-completions.xsh
│   └── utils.xsh
├── rc.xsh
```

The exceptions thrown when I have carapace in my `rc.xsh` suggested that Xonsh is sourcing foreign shell run control files, though I haven't asked it to do so. To test that theory, I added `echo "this is <whatever>"` to other shell run control files. When `~/.bashrc` is available and carapace is not enabled in `rc.xsh`, launching an iTerm2 profile that starts `~/.local/xonsh-env/xbin/xonsh` will print `this is bashrc`, and everything will start up just fine. Note that it also prints that when running `xonsh --no-rc` from another shell.

When `~/.bashrc` is not available and carapace is disabled in `rc.xsh`, Xonsh throws exceptions indicating that `starship` and `zoxide` — both initialized in `rc.xsh` aren't available. They're both in `/opt/homebrew/bin`, which is added to PATH in `.bashrc` and in `~/.config/xonsh/rc.d/00-path.xsh`, so that tells me that Xonsh sources `rc.xsh` first. That's fine, I can move the stuff in `00-path.xsh` to `rc.xsh`. I do occasionally drop to Bash, so `~/.bashrc` needs to stay put.

But why is `.bashrc` being sourced in the first place? I get that it's a convenience feature to be able to source a foreign shell config, and that's lovely if you want it, but I don't, and I can't find any documentation on how to disable it. Everything I've read seems to suggest that it's opt-in, and I'm about 99% sure I didn't.

So, questions: How can I tell for sure whether I've configured Xonsh to source `~/.bashrc`, and how can I configure it to no longer do so? TIA 🙏🏽

**Comment by anki-code** [comment, marked as answer]
I've created an issue #5768. Thank you for reporting this.
Historically I think the idea was to have seamless support of features/aliases from another shells. But now xonsh is much stronger and we need to rethink this and make foreign RC loading optional. PR with disabling this is welcome!
